device/transport/mqtt/src/mqtt.ts

```diff
@@ -672,24 +660,81 @@ export class Mqtt extends EventEmitter implements Client.Transport {
     this._fsm.handle('disableTwin', callback);
   }
 
+  /**
+   * @private
+   */
+  sendOutputEvent(outputName: string, message: Message, done: (err?: Error, result?: results.MessageEnqueued) => void): void {
+    debug('sendOutputEvent ' + JSON.stringify(message));
+
+    let topic = this._getEventTopicFromMessage(message, { '$.on': outputName });
+    topic += '/';
+
+    this._fsm.handle('sendEvent', topic, message.data, { qos: 1, retain: false }, (err, puback) => {
+      if (err) {
+        debug('send error: ' + err.toString());
+        done(err);
+      } else {
+        debug('PUBACK: ' + JSON.stringify(puback));
+        done(null, new results.MessageEnqueued(puback));
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  sendOutputEventBatch(outputName: string, messages: Message[], done: (err?: Error, result?: results.MessageEnqueued) => void): void {
+    throw new errors.NotImplementedError('MQTT Transport does not support batching yet');
```

Comment:
is there a plan for batching with other protocols?

Reply:
investigating.  If not, we should remove the API

---
In reply to: [170389068](https://github.com/Azure/azure-iot-sdk-node/pull/235#discussion_r170389068) [](ancestors = 170389068)